### ISSUE TITLE
Proposal: logging methods should always be bound to `console`

### ISSUE THREAD

**Issue by fitzgen** [DeveloperToolsWG/console-object#27]
```
[1,2,3].forEach(console.log)
```

Expected Results:

logs 3 messages in the console

Actual Results:

Error: 'log' called on an object that does not implement interface Console.

---

This used to be the behavior in Firefox, but it recently broke. I've filed a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=989619) to revert to the old behavior where the methods were always bound to console, but figured we should codify this behavior here too.

Its kind of masochistic to force developers to bind every time they pass `console.log` as an argument!

**Comment by jamiesonbecker** [comment]
I agree - I personally use this model all the time; for example, I'll often replace a callback function name with console.log during development (usually in Firebug). Writing a hand-off function is a big pain in the console. :)

**Comment by sindresorhus** [comment]
:+1: It's very convenient when you want to log the output of a callback `.catch(console.error)` or create a shortcut `const log = console.log`. This is also how it works in Node.js.
